### PR TITLE
RSDK-6633: Derive get_images timestamps from DepthAI directly instead of time.time() inside our module

### PR DESCRIPTION
**Pull request by hexbabe**
https://viam.atlassian.net/browse/RSDK-6633

Sorry for the confusing commit history! I made this branch off of a feature branch instead of main on accident. Please look at the final diff instead of commit-by-commit.

The main goal of this PR is to get rid of `time.time()` calls for timestamps. We want to use DepthAI's `get_timestamp` for better accuracy.

There are also some readability fixes.

### CODE REVIEW

File: src/worker.py [hunk 1]
```diff
@@ -156,26 +156,20 @@ def __init__(
...
-            depth_output, timestamp
-        )
-
     def get_color_image(self) -> Optional[CapturedData]:
         color_q = self.color_q_handler.get_queue()
         try:
             color_msg = color_q.get(block=True, timeout=5)
             color_output = self._process_color_frame(color_msg.frame)
-            timestamp = time.time()
+            timestamp = color_msg.get_timestamp().total_seconds()
```
Comment (hexbabe) [inline src/worker.py:172]: This + similar are the key changes made.

File: src/worker.py [hunk 2]
```diff
@@ -185,7 +179,7 @@ def get_depth_map(self) -> Optional[CapturedData]:
         try:
             depth_msg = depth_q.get(block=True, timeout=5)
             depth_output = self._process_depth_frame(depth_msg.frame)
-            timestamp = time.time()
+            timestamp = depth_msg.get_timestamp().total_seconds()
```
Comment (hexbabe) [inline src/worker.py:182]: Here too

File: src/worker.py [hunk 3]
```diff
@@ -198,7 +192,7 @@ def get_pcd(self) -> CapturedData:
                 pc_output = self._downsample_pcd(pc_msg.points, pc_msg.points.nbytes)
             else:
                 pc_output = pc_msg.points
-            timestamp = time.time()
+            timestamp = pc_msg.get_timestamp().total_seconds()
```
Comment (hexbabe) [inline src/worker.py:195]: And here

File: src/worker.py [hunk 4]
```diff
@@ -229,12 +223,14 @@ def _try_get_synced_color_depth_outputs(
...
-            color_frame = synced_color_msgs["color"].frame
-            depth_frame = synced_color_msgs["depth"].frame
-
-            color_output = self._process_color_frame(color_frame)
-            depth_output = self._process_depth_frame(depth_frame)
-            return color_output, depth_output
+            color_frame, depth_frame, timestamp = (
+                synced_color_msgs["color"].frame,
+                synced_color_msgs["depth"].frame,
+                synced_color_msgs["color"].get_timestamp().total_seconds(),
```
Comment (hexbabe) [inline src/worker.py:229]: And finally here

**Review by seanavery — APPROVED** [review]
LGTM.
Nice catch/update!

**Pull request merged**